src/syscheckd/run_check.c

```diff
@@ -351,42 +351,24 @@ void start_daemon()
 }
 // LCOV_EXCL_STOP
 
-
-// LCOV_EXCL_START
 // Starting Real-time thread
-#ifdef WIN32
+#if defined WIN32
 DWORD WINAPI fim_run_realtime(__attribute__((unused)) void * args) {
-#else
-void * fim_run_realtime(__attribute__((unused)) void * args) {
-#endif
-
-#if defined INOTIFY_ENABLED || defined WIN32
-    static int _base_line = 0;
-#ifdef WIN32
     directory_t *dir_it;
-
+    int _base_line = 0;
     set_priority_windows_thread();
-#endif
-
-    while (1) {
-#ifdef WIN32
+    while (FOREVER()) {
         // Directories in Windows configured with real-time add recursive watches
         foreach_array(dir_it, syscheck.directories) {
             if (dir_it->options & REALTIME_ACTIVE) {
                 realtime_adddir(dir_it->path, dir_it, 0);
             }
         }
-#endif
 
         if (_base_line == 0) {
             _base_line = 1;
-
             if (syscheck.realtime != NULL) {
-                if (syscheck.realtime->queue_overflow) {
-                    realtime_sanitize_watch_map();
-                    syscheck.realtime->queue_overflow = false;
-                }
-                mdebug2(FIM_NUM_WATCHES, syscheck.realtime->dirtb->elements);
+                mdebug2(FIM_NUM_WATCHES, OSHash_Get_Elem_ex(syscheck.realtime->dirtb));
             }
         }
```

Comment:
Could we move this whole block outside of the `while` loop and get rid of `_base_line`?